enhancements/security/centralized-tls-config.md

```diff
@@ -0,0 +1,397 @@
+- Inconsistent TLS behavior across the platform
+- Inability to enforce custom TLS profiles defined by customers
+- Blocked path to PQC readiness, as PQC-resilient algorithms will only be available in TLS 1.3+
+- Customer security compliance failures when components ignore configured cipher restrictions
+
+### User Stories
+
+#### Story 1: Security Administrator Enforcing TLS Policy
+As a security administrator, I want all OpenShift platform components to respect the TLS security profile I configure at the cluster level so that I can ensure consistent cryptographic policy enforcement across my entire cluster without having to configure each component individually.
+
+#### Story 2: PQC Preparation
+As a cluster administrator preparing for Post-Quantum Cryptography requirements, I want to enforce TLS 1.3 across my entire platform so that when PQC algorithms become available (via the separate curves configuration enhancement), all components will be ready to use them consistently.
+
+#### Story 3: Custom TLS Profile Enforcement
+As a security-conscious enterprise customer, I want to define a custom TLS profile that disables algorithms my security team considers unsafe, and I want all OpenShift components (including operators and layered products) to honor this configuration so that I can meet my organization's security compliance requirements.
+
+#### Story 4: Upgrade Safety
+As a cluster administrator, I want to have control over when components begin honoring centralized TLS configuration after an upgrade so that I can test the impact on my workloads and clients before fully enabling strict TLS policy enforcement.
+
+### Goals
+
+1. All OpenShift core platform components honor the centralized TLS configuration from the API Server, Kubelet, or Ingress configuration (as appropriate for each component).
+2. Layered products and operators also respect the centralized TLS configuration.
+3. Components explicitly configure all TLS profile settings rather than relying on Go/library defaults.
+4. Provide a safe upgrade path that allows customers to gradually adopt stricter TLS enforcement without breaking existing workloads or clients.
+
+### Non-Goals
+
+1. Modifying the existing TLS security profile API structure (Old, Intermediate, Modern, Custom profiles) - that API is already defined.
+2. Adding the `curves` field to the TLS profile API - that is covered by a separate enhancement (see [api-tls-curves-config.md](/enhancements/security/api-tls-curves-config.md)).
+3. Defining specific PQC algorithms or curves to be used - this enhancement focuses on the enforcement mechanism.
+4. Changing how TLS certificates are managed or distributed.
+
+## Proposal
+
+This proposal introduces a mechanism for gradual adoption of centralized TLS configuration enforcement, allowing components to transition from their current behavior (often ignoring the central config) to fully respecting the configured TLS profile.
+
+### TLS Adherence Mode
```

Comment:
As discussed in our sync, I think I would suggest swapping this proposal to an alternative, and making the main proposal be the introduction of a new overarching API that (when present) would dictate the desired TLS configuration of all cluster components (or at least all non-ingress-related components? I'm not sure we fully ran that aspect of the conversation to ground)